msbuild.js

```diff
@@ -123,7 +123,42 @@ msbuild.prototype.getMSBuildPath = function(os,processor,version){
 	if(os === 'linux' || os === 'darwin') return "xbuild";
 	
 	var frameworkDirectories,programFilesDir,msbuildDir,exeDir;
+	var vs2017Type = {
+		Pro: 'Professional', 
+		Enterprise: 'Enterprise', 
+		Community: 'Community', 
+		BuildTools: 'BuildTools'
+	};
+
 	programFilesDir = process.env['programfiles(x86)'] || process.env.PROGRAMFILES;
+
+	// For the msbuild 15.0 version, use the appropriate VS2017 directories
+	if (version === "15.0") {
+
+		// If VSINSTALLDIR env. var cannot be found, see what could be the directory by searching the usual suspects
+		// (while giving higher priority to the VS2017 IDE installs over the Build Tools only install)
+		if (process.env.vsInstallDir === undefined) {
```

Comment:
Sorry I do not see the difference from my code. Lines 136 to 140 are same as I had them... unless I am missing something.